client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import Search from 'components/search';
+import SegmentedControl from 'components/segmented-control';
+import { trackClick } from '../helpers';
+import config from 'config';
+import { isMobile } from 'lib/viewport';
+
+const ThemesMagicSearchCard = React.createClass( {
+	propTypes: {
+		tier: React.PropTypes.string,
+		select: React.PropTypes.func.isRequired,
+		site: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		onSearch: React.PropTypes.func.isRequired,
+		search: React.PropTypes.string
+	},
+
+	trackClick: trackClick.bind( null, 'search bar' ),
+
+	componentWillMount() {
+		this.onResize = debounce( () => {
+			this.setState( { isMobile: isMobile() } );
+		}, 250 );
+	},
+
+	componentDidMount() {
+		window.addEventListener( 'resize', this.onResize );
+	},
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.onResize );
+	},
+
+	getInitialState() {
+		return {
+			isMobile: isMobile(),
+			searchIsOpen: false
+		};
+	},
+
+	getDefaultProps() {
+		return { tier: 'all' };
+	},
+
+	onSearchOpen() {
+		this.setState( { searchIsOpen: true } );
+	},
+
+	onSearchClose() {
+		this.setState( { searchIsOpen: false } );
+	},
+
+	onBlur() {
+		if ( this.state.isMobile ) {
+			this.setState( { searchIsOpen: false } );
+		}
+	},
+
+	render() {
+		const isJetpack = this.props.site && this.props.site.jetpack;
```

Comment:
> It's preferable for `var`, `const`, and `let` statements to stay each one on an individual line

That's from the updated document. It's a "preference", so if one feels strongly to do otherwise, it's ok. My advice would be to try to have **one clear statement per line**, for a variety of reasons that however it's probably better to not discuss here. ;)